cedux.h

```diff
@@ -9,11 +9,20 @@
                                                                                                 \
 struct STORE_NAME##_handle;                                                                     \
 typedef void(*STORE_NAME##_REDUCER)(TREE_TYPE * p_tree, ACTION_TYPE action);                    \
+typedef void(*STORE_NAME##_SUBSCRIBER)(const TREE_TYPE * p_tree, void *data);                   \
+struct STORE_NAME##_subscriber_container                                                        \
+{                                                                                               \
+    STORE_NAME##_SUBSCRIBER subscriber;                                                         \
+    void *data;                                                                                 \
+};                                                                                              \
 QUEUE_DECLARATION(STORE_NAME##_action_queue, ACTION_TYPE, 16)                                   \
 LIST_DECLARATION(STORE_NAME##_reducer_list, STORE_NAME##_REDUCER, 32)                           \
+LIST_DECLARATION(STORE_NAME##_subscriber_list, struct STORE_NAME##_subscriber_container *, 32)  \
```

Comment:
Do you feel strongly that this list should contain pointers to the subscriber containers rather that just the structs themselves? Since the container is really just two pointers they don't take up much space. In your example, the subscriber containers are declared on the stack in the main function, which works well in that case because they will never go out of scope, however, if you wanted to add subscribers from another function elsewhere in the app, you would have to declare them statically somewhere. To reduce confusion, I would probably vote to just store the struct in the list.

Reply:
Regarding the `void *data` field, I'm on the fence as well. I agree it could be useful in some cases where you want to have access to some state that you don't want to put in the store. You don't have any type safety of course but does provide some additional flexibility. I don't mind keeping it there.

Reply:
Yeah, I was concerned about those scope issues as well.  As long as the LIST can handle the struct itself instead of a pointer, I agree that would be much simpler.  Ideally the library user wouldn't even know about the container struct.  Let me give it a shot after I fix the merge conflicts.

(TBH, I've been passing structs by pointer for so long now (decades?) that I had forgotten you _could_ pass them by value.)